enhancements/update/standalone-worker-surge.md

```diff
@@ -0,0 +1,295 @@
+
+#### Replace Update Type
+
+The `Replace` update type removes old machines and replaces them with new instances. It supports
+two strategies:
+- `OnDelete` where a new machine is brought online only after the old machine is deleted.
+- `RollingUpdate` which supports the `MaxSurge` option.
+
+```yaml
+kind: MachineConfigPool
+spec:
+  # Existing spec fields are not shown.    
+
+  machineManagement:
+    upgradeType: "Replace"
+    replace:
+      strategy: "RollingUpdate"
+      rollingUpdate:
+        maxUnavailable: 0
+        maxSurge: 4
+```
+
+`MaxSurge` applies independently to each associated MachineSet. For example, if three MachineSets are associated
+with a MachineConfigPool, and `MaxSurge` is set to 4, then it is possible for the cluster to surge up to 12 nodes
+(4 for each of the 3 MachineSets).
+
+### Topology Considerations
+
+Multi-AZ (availability zone) clusters function by using one or more `MachineSets` per zone. In order for this enhancement
+to work across all zones, all such `MachineSets` should be associated with a `MachineConfigPool` with well considered
+values for `MaxSurge` and `NodeDrainTimeout`.
+
+Each `MachineSet` associated with a `MachineConfigPool` will be permitted to scale by the `MaxSurge` number of nodes. 
+The alternative (trying to spread a surge value evenly across `MachineSets`) is problematic. Consider a cluster with two `MachineSets`:
+- machine-set-1a which creates nodes in availability zone us-east-1a.
+- machine-set-1b which creates nodes in availability zone us-east-1b.
+
+Further, assume that `MaxSurge` is set to 1 for the `MachineConfigPool` associated with these `MachineSets`.
+
+There may be pods running on 1b nodes that can only be scheduled on 1b nodes (e.g. due to taints / affinity / 
+topology constraints, machine type, etc.). If `MaxSurge` was interpreted in such a way as to only surge machine-set-1a by 1 node,
+constrained pods requiring 1b nodes could not benefit from this additional capacity.
+
+Instead, this enhancement proposes each `MachineSet` be permitted to surge up to the `MachineConfigPool` surge
+value independently. 
+
+
+#### Hypershift / Hosted Control Planes
+
+N/A. Hosted Control Planes provide the model for the settings this enhancement seeks to expose in standalone clusters.
+
+#### Standalone Clusters
+
+The `MachineConfigPool` custom resource must be updated to expose the new semantics.  The existing `spec.maxUnavailable`
+will be deprecated in favor of the more expressive `MachineManagement` stanza. 
+
+#### Single-node Deployments or MicroShift
+
+N/A.
+
+### Implementation Details/Notes/Constraints
+
+`MachineSets` or CAPI equivalents must support `MachineConfigPool` values for `MaxSurge`. When `MaxSurge` is greater
+than 0, the controller can instantiate more machines than the number of specified `MachineSet` replicas. 
+This must work seamlessly with all cluster autoscaler options. The cluster autoscaler may need to be aware of 

```

Comment:
MUO side-steps this by creating new machinesets for the surge capacity.

Reply:
Resolving since I've added this as the proposed implementation.